test/element.test.js

```diff
@@ -75,6 +75,97 @@ describe('Element', function() {
     });
   });
 
+  describe('#at', function() {
+    context('when the element is not expandable', function() {
+      var element = new Element('name', 'test');
+
+      it('returns undefined', function() {
+        expect(element.at(0)).to.equal(undefined);
+      });
+    });
+
+    context('when the element exists for the index', function() {
+      var element = new Element('key', ['item0', 'item1']);
+
+      it('returns the element', function() {
+        expect(element.at(0).currentValue).to.equal('item0');
+        expect(element.at(1).currentValue).to.equal('item1');
+      });
+    });
+
+    context('when the element exists but one has been removed', function() {
+      var element = new Element('key', ['item0', 'item1']);
+
+      before(function() {
+        element.at(0).remove();
+      });
+
+      it('returns the original elements', function() {
+        expect(element.at(0).currentValue).to.equal('item0');
+        expect(element.at(0).currentKey).to.equal(0);
+        expect(element.at(1).currentValue).to.equal('item1');
+        expect(element.at(1).currentKey).to.equal(1);
+      });
+    });
+
+    context('when the element is deleted', function() {
+      var element = new Element('key', []);
+      var child = element.insertEnd('', 'test');
+
+      before(function() {
+        child.remove();
+      });
+
+      it('returns undefined', function() {
+        expect(element.at(0)).to.equal(undefined);
+      });
+    });
+
+    context('when the element is deleted from the middle', function() {
+      var element = new Element('key', []);
+      var child1 = element.insertEnd('', 'test1');
+      element.insertEnd('', 'test2');
+      var child2 = element.insertEnd('', 'test3');
+      element.insertEnd('', 'test4');
+      var child3 = element.insertEnd('', 'test5');
+
+      before(function() {
+        child1.remove();
+        child2.remove();
+        child3.remove();
+      });
+
+      it('returns undefined', function() {
```

Comment:
I think this description needs to be changed to what is actually expected. The tests aren't looking for `undefined` values except for the last element.